codegen/directive.go

```diff
@@ -172,3 +178,23 @@ func (d *Directive) Declaration() string {
 	res += ") (res interface{}, err error)"
 	return res
 }
+
+func (d *Directive) IsBuiltIn() bool {
+	return d.Implementation != nil
+}
+
+func (d *Directive) CallPath() string {
+	if d.IsBuiltIn() {
+		return "builtInDirective" + d.CallName()
```

Comment:
We now have two types of directives: user provided and "built in" where the implementation is statically provided by the plug in.

For `builtIn` directives we call a static function vs an instance function.